frontend-react/src/hooks/UseValueSets.ts

```diff
@@ -0,0 +1,152 @@
+import axios from "axios";
+import { useQuery } from "@tanstack/react-query";
+
+import {
+    LookupTable,
+    lookupTableApi,
+    LookupTables,
+    ValueSet,
+    ValueSetRow,
+} from "../network/api/LookupTableApi";
+
+export interface TableAttributes {
+    version: number;
+    createdAt?: string;
+    createdBy?: string;
+}
+
+/*
+
+  Fetch / Mutate functions and helpersa
+
+*/
+
+/*
+TO DISCUSS: we could set the pattern to be something like this or
+we could reference query keys within the function https://stackoverflow.com/a/68111112/5500298
+
+Have not tested, but I feel like to get the full value of react-query's refetching behavior we may want to go with the last option.
+When react-query refetches it will just re-run the function we passed to it, this would not take into account changes in variable dependencies
+unless we bake those into function itself?
+
+Note that I ran into type errors when trying to use higher order functions to avoid the inlining
+*/
+
+// will return either a value set or value set row depeding on
+// value set table name
+// we may not need the generic here, react-query probably handles that. But it couldn't hurt?
+const getLatestData = <T>(tableName: LookupTables, version: number) => {
+    const endpointHeader = lookupTableApi.getTableData<T>(version, tableName);
+
+    return axios
+        .get<T>(endpointHeader.url, endpointHeader)
+        .then((response) => response.data);
+};
+
+// not going to type this yet, as this function will likely go away when we bring in fetch by provider
```

Comment:
> ...when we bring in fetch by provider

Can you elaborate? I'm blanking on "fetch by provider"

Reply:
yeah, that is some confusing short hand there. I'm planning in PR 3 (or 4, technically, with the app wrapper thing in the mix now), to introduce a provider that will `provide` access to a universal fetch function. Mainly meant to mention that it's not worth spending too much time worrying about this function as it will go away soon.

Reply:
Oh, that's super cool! Good to know.